### PR TITLE
DNN-8513: Create a new extension point in the DNN core at the Database Connection Level.

### PR DESCRIPTION
**Pull request by JoshuaBradley**
Can use this package for testing:[https://github.com/JoshuaBradley/DnnAzureSqlConnectionProvider](https://github.com/JoshuaBradley/DnnAzureSqlConnectionProvider)

### CODE REVIEW

File: DNN Platform/Library/Data/DatabaseConnectionProvider.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+#region usings
+
+using System;
+using System.Linq;
+
+#endregion
+
+namespace DotNetNuke.Data
+{
+    using System.Data;
```
Comment (bdukes) [inline DNN Platform/Library/Data/DatabaseConnectionProvider.cs]: Can you move all `using` statements into `#region usings` above?

File: DNN Platform/Library/Data/SqlDatabaseConnectionProvider.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    public class SqlDatabaseConnectionProvider : DatabaseConnectionProvider
+    {
+        public override void ExecuteCommand()
+        {
+            //Create a new connection
+            using (var connection = new SqlConnection(ConnectionString))
+            {
+                //Create a new command (with no timeout)
+                var command = new SqlCommand(Query, connection) { CommandTimeout = 0 };
```
Comment (bdukes) [inline DNN Platform/Library/Data/SqlDatabaseConnectionProvider.cs]: Is this supposed to override/ignore the `CommandTimeout` property?  The [calling code](#diff-ac4f83a40d8acd75b4980dfa00e3d574R115) sets `CommandTimeout` to `0`, should this be done from the calling code, or always set here in the provider?  Off the top of my head, I would assume the calling code should be responsible for setting the timeout.

**Comment by bdukes** [comment]
@JoshuaBradley are you alright with distributing your [DnnAzureSqlConnectionProvider](https://github.com/JoshuaBradley/DnnAzureSqlConnectionProvider) as an optional install with DNN Platform?

**Comment by JoshuaBradley** [comment]
I have made the changes suggested. I am fine with the [DnnAzureSqlConnectionProvider](https://github.com/JoshuaBradley/DnnAzureSqlConnectionProvider) being included as an optional install package and would like to ultimately have the project be included in the platform.

**Comment by bdukes** [comment]
@wmorgenweck @jbrinkman @ashishpd is there interest in this feature?

**Comment by wmorgenweck** [comment]
This is very interesting.  @ashishpd, can we have engineer review this?  Can we get @davidjrh to take a looks as well?

**Comment by bdukes** [comment]
Any feedback/status, @wmorgenweck or @ashishpd?

**Comment by galatrash-at-dnn** [comment]
@JoshuaBradley - this is an old PR which has lots of conflicts with the current source code. I went through it and it looks good except for:
- the conflicts,
- basing on old release (DNN Platform/Website/Install/Config/08.00.02.01.config)
- the new time-out parameter that I introduced a while ago.

Attached is a copy of the merged SqlDataProvider.cs file to the Jira issue -
 https://dnntracker.atlassian.net/browse/DNN-8513 - which you can use as a base to modify your PR to get your code in-line with the current. Also, you need to align the other artifacts to the new release.

As we are close to code freeze and this is a big core change, I suggest you keep the new PR till the release that comes after 9.2.0 (preferably at the beginning of the release to give enough time for testing).

Of course, you need to update the connector library you attached to the Jira issue as well.

**Comment by galatrash-at-dnn** [comment]
@JoshuaBradley any plans to work on this soon? Please let me know.

**Comment by galatrash-at-dnn** [comment]
Base on no response from the original submitter, I will close this PR without merging. I will create a new PR for this with all suggested changes.

**Pull request closed without merging**